core/bin/migrations/20240603_backfill_stored_documents.rs

```diff
@@ -0,0 +1,356 @@
+use ::qdrant_client::qdrant::{value, PointId};
+use anyhow::{anyhow, Context, Result};
+use cloud_storage::Object;
+use dust::{
+    data_sources::{
+        data_source::{make_document_id_hash, DataSource, Section},
+        file_storage_document::FileStorageDocument,
+        qdrant::QdrantClients,
+    },
+    stores::{postgres, store},
+    utils,
+};
+use qdrant_client::qdrant;
+use tokio_postgres::Row;
+
+use bb8::Pool;
+use bb8_postgres::PostgresConnectionManager;
+use futures::{StreamExt, TryStreamExt};
+use tokio_postgres::NoTls;
+use tokio_stream::{self as stream};
+
+async fn fetch_data_sources_batch(
+    pool: &Pool<PostgresConnectionManager<NoTls>>,
+    last_id: u64,
+    limit: usize,
+) -> Result<Vec<Row>, anyhow::Error> {
+    let c = pool.get().await?;
+
+    c.query(
+        "SELECT id, internal_id FROM data_sources WHERE id > $1 LIMIT $2",
+        &[&(last_id as i64), &(limit as i64)],
+    )
+    .await
+    .context("Query execution failed")
+}
+
+// Return a "best-effort" section for the latest version of the document (based on Qdrant).
+// For superseded versions, return an empty section.
+async fn get_sections_from_document_version(
+    qdrant_clients: &QdrantClients,
+    ds: &DataSource,
+    document_id_hash: &str,
+    document_status: &str,
+) -> Result<Section> {
+    if document_status != "latest" {
```

Comment:
In `FileStorageDocument` I set `Section` to not optional https://github.com/dust-tt/dust/blob/d7cca388cdd0eb8a4d8002bf471f996e4f90e8ab/core/src/data_sources/file_storage_document.rs#L12-L16.

Hence the workaround here. I don't want to include the entire document text because I want to ensure that each section adheres to the chunk logic. IMHO, it's acceptable since these files will never be used for reindexing. These versions have all been superseded and no longer exist in Qdrant.

Reply:
My FUD here is that it we ever end up wanting to do anything with this data then we will be likely do the wrong thing no?

Why don't you want to include the full text. If we ever use the section well do something sensible with it?

Reply:
Section capture structure but don't need to be chunked in the right size.

So it's actually very valid to have one big section with no structure.

Reply:
Thanks! LGTM generally. Happy to make a final pass when needed :+1:

Reply:
Please do, I'd like to start the migration today.